gamemodes/terrortown/gamemode/shared/sh_sprint.lua

```diff
@@ -87,11 +87,13 @@ hook.Add("Think", "TTT2PlayerSprinting", function()
 		if not ply.sprintTS then continue end
 
 		local timeElapsed = CurTime() - ply.sprintTS
-
-		if not ply.sprintMultiplier then
-			ply.sprintProgress = math.min((ply.oldSprintProgress or 0) + timeElapsed * GetGlobalFloat("ttt2_sprint_stamina_regeneration"), 1)
+		local drainMult = hook.Run("TTT2StaminaDrain", ply)
```

Comment:
These hooks should be placed in the right if-clause to avoid the call if it's not necessary :)